clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -252,6 +252,33 @@ public synchronized void update(Cluster newCluster, Set<String> unavailableTopic
         if (isClosed())
             throw new IllegalStateException("Update requested after metadata close");
 
+        String previousClusterId = cluster.clusterResource().clusterId();
+        String newClusterId = newCluster.clusterResource().clusterId();
+
+        if (previousClusterId != null && !newClusterId.equals(previousClusterId)) {
+            // kafka cluster id is unique.
+            // On client side, the cluster id in Metadata is only null during bootstrap, and client is
+            // expected to talk to the same cluster during its life cycle. Therefore if cluster id changes
+            // during metadata update, meaning this metadata update response is from a different cluster,
+            // client should reject this response and not update cached cluster to the wrong cluster
+
+            // According to SRE, removing brokers and adding to another cluster is common operation, thus
```

Comment:
this paragraph may be too linkedin specific ?

Reply:
Sure, will rephrase this.